src/build-data/ninja.in

```diff
@@ -10,7 +10,7 @@ LANG_EXE_FLAGS = %{cc_lang_binary_linker_flags}
 CXXFLAGS       = %{cc_compile_flags}
 WARN_FLAGS     = %{cc_warning_flags}
 
-LDFLAGS        = %{ldflags} %{cc_compile_flags}
+LDFLAGS        = %{ldflags}
```

Comment:
This behavior (passing the compiler flags also to the linker) was just added in #4206 which I guess is kind of inelegant but we'd need to figure out some alternate solution for #4196